### ISSUE TITLE
ServerVersion could contains error text if select version() fails

### ISSUE THREAD

**Issue by pprogs** [DarkWanderer/ClickHouse.Client#55]
If there is an error in reading version, than ServerVersion will contains error text and will fail at following Version.Parse calls.
maybe try parse version in [openAsync](https://github.com/DarkWanderer/ClickHouse.Client/blob/e03a67b442a763181a2e8fa256a5878befa8997b/ClickHouse.Client/ADO/ClickHouseConnection.cs#L246) call

**Comment by DarkWanderer** [comment]
ServerVersion must not contain error - if it does, it means the connection was not initialized properly (as version is required for feature toggles). Can you tell me how you receive this error?

**Comment by pprogs** [comment]
I`ve got this error when server was low on memory and no queries could be run.
i've put version parse afeter openAsync and got this
```C#
try
{
    Version.Parse(con.ServerVersion);
}
catch (Exception ex)
{
    throw new Exception(
        $"Could not parse server version from clickhouse! ver ({con.ServerVersion}). {ex.Message}");
}
```
"Could not parse server version from clickhouse! ver (Code: 241, e.displayText() = DB::Exception: Memory limit (total) exceeded: would use 3.47 GiB (attempt to allocate chunk of 4256108 bytes), maximum: 3.47 GiB: While executing TabSeparatedRowOutputFormat (version 20.10.3.30 (official build))). Version string portion was too short or too long. (Parameter 'input')"

**Comment by DarkWanderer** [comment]
Ok, I see your point - this error should have been indicated earlier, will fix

**Comment by DarkWanderer** [comment]
What I find weird, however, is that there should have been an exception a few lines earlier in `OpenAsync`, because the error should not have had a successful return code. Do you have any other code modifications there?

**Comment by pprogs** [comment]
Nope, just this
```C#
var con = new ClickHouseConnection(_connectionString);
await con.OpenAsync(token);
if (con.State != ConnectionState.Open)
    throw new Exception($"Could not open clickhouse connection. ver ({con.ServerVersion})");
try
{
    Version.Parse(con.ServerVersion);
}
catch (Exception ex)
{
    throw new Exception(
        $"Could not parse server version from clickhouse! ver ({con.ServerVersion}). {ex.Message}");
}
```

**Comment by DarkWanderer** [comment]
Ok, must be a ClickHouse issue then - it should have returned a non-successful HTTP code.
In this case looks like it didn't, otherwise `response.EnsureSuccessStatusCode();` would have thrown.

Will be fixed in next release

**Comment by pprogs** [comment]
ok, thanks!

**Comment by DarkWanderer** [comment]
Fixed in [3.0](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/3.0.0.357)
